File: leetcode/#867.js
// question --> https://leetcode.com/problems/transpose-matrix/

/**
 * @param {number[][]} matrix
 * @return {number[][]}
 */
var transpose = function(matrix) {
    let rows = matrix.length;
    let cols = matrix[0].length;
    let ans = new Array();
    for (let c = 0; c < cols; c++) {
        ans[c] = Array(rows); 
    }
    
    for (let c = 0; c < cols; c++) {
        for (let r = 0; r < rows; r++) {
            ans[c][r] = matrix[r][c];
        }
    }
    return ans;
};

// TC: O(N); SC: O(N) 

// end

File: leetcode/#867.py
# question --> https://leetcode.com/problems/transpose-matrix/

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rows, cols = len(matrix), len(matrix[0])
        ans = [[0 for r in range(rows)] for c in range(cols)]
        for c in range(cols):
            for r in range(rows):
                ans[c][r] = matrix[r][c] 
        return ans  
        
        # TC: O(R*C); SC: O(R*C)
        
# Editorial

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        return zip(*matrix)
        
# end
